open-api/rest-catalog-open-api.yaml

```diff
@@ -2838,6 +2978,63 @@ components:
           additionalProperties:
             type: string
 
+    PreplanTableRequest:
+      type: object
+      required:
+        - select
+        - filter
+        - case-sensitive
+      properties:
+        select:
+          description: A list of the selected column names
+          type: array
+          items:
+            type: string
+        filter:
+          $ref: '#/components/schemas/Expression'
+        case-sensitive:
+          description: Indicates whether column selection and filtering should be case sensitive
+          type: boolean
+        snapshot-id:
+          description: an int64 snapshot ID (if snapshot-range is not present); optional and defaults to the table's current snapshot
+          type: integer
+          format: int64
+        snapshot-range:
+          description: a JSON list containing exactly 2 int64 snapshot IDs (if snapshot-id is not present) representing the start (defaults to exclusive, see `start-exclusive`) and end (only inclusive) snapshots
+          type: array
+          items:
+            type: integer
+            format: int64
+        start-exclusive:
+          description: Indicates whether the user wants the start of `snapshot-range` to be inclusive or exclusive
+          type: boolean
+          default: true
+
+    PlanTableRequest:
+      type: object
+      required:
+        - select
+        - filter
+        - case-sensitive
+      properties:
```

Comment:
Why not allow `snapshot-id` and `snapshot-range` options here?

Reply:
I think there was a comment left a while back mentioning that if a user passes a `plan-task` in the `PlanTableRequest` that this conflicts if the `snapshot-id` and `snapshot-range` are also passed(as the plan task could done on different snapshot). 

But if a user does not pass `plan-task` then im assuming we want to have these options available?

Reply:
Yes. If there is no `plan-task` then those options should be available.

Reply:
will fix this